packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -0,0 +1,124 @@
+ * @param thumbProps Props used for this specific thumb.
+ * @param state Slider state, created via `useMultiSliderState`.
+ * @param trackRef Ref for the track element.
+ * @param inputRef Ref for the range input element.
+ */
+export function useSliderThumb(
+  sliderProps: BaseSliderProps,
+  thumbProps: SliderThumbProps,
+  state: MultiSliderState,
+  trackRef: React.RefObject<HTMLElement>,
+  inputRef: React.RefObject<HTMLInputElement>
+): SliderThumbAria {
+  const {
+    maxValue = DEFAULT_MAX_VALUE,
+    minValue = DEFAULT_MIN_VALUE,
+    step = DEFAULT_STEP_VALUE,
+    isReadOnly: isSliderReadOnly,
+    isDisabled: isSliderDisabled
+  } = sliderProps;
+
+  const {
+    index,
+    isRequired,
+    isDisabled,
+    isReadOnly,
+    validationState,
+    labelId
+  } = thumbProps;
+
+  const {labelProps, fieldProps} = useLabel({
+    ...thumbProps,
+    'aria-labelledby': `${labelId} ${thumbProps['aria-labelledby'] ?? ''}`.trim()
+  });
+
+  const value = state.values[index];
+  const allowDrag = !(isSliderDisabled || isSliderReadOnly || isDisabled || isReadOnly);
+
+  const focusInput = useCallback(() => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputRef]);
+
+  const isFocused = state.focusedIndex === index;
+
+  useEffect(() => {
+    if (isFocused) {
+      focusInput();
+    }
+  }, [isFocused, focusInput]);
+
+  const draggableProps = useDrag1D({
+    containerRef: trackRef as any,
+    reverse: false,
+    orientation: 'horizontal',
+    onDrag: (dragging) => {
+      state.setDragging(index, dragging);
+      console.log('THUMB FOCUS');
+      focusInput();
+    },
+    onPositionChange: (position) => {
+      state.setValue(index, computeOffsetToValue(position, sliderProps, trackRef));
+    },
+    onIncrement: () => state.setValue(index, value + step),
+    onDecrement: () => state.setValue(index, value - step),
+    onIncrementToMax: () => state.setValue(index, maxValue),
+    onDecrementToMin: () => state.setValue(index, minValue)
+  });
+
+  const {focusableProps} = useFocusable(
+    mergeProps(thumbProps, {
+      onFocus: () => state.setFocusedIndex(index),
+      onBlur: () => state.setFocusedIndex(undefined)
+    }),
+    inputRef
+  );
+
+  return {
+    inputProps: mergeProps(focusableProps, fieldProps, {
+      type: 'range',
+      tabIndex: allowDrag ? 0 : undefined,
+      'aria-valuenow': value,
+      'aria-valuemin': state.getMinValueForIndex(index),
+      'aria-valuemax': state.getMaxValueForIndex(index),
+      'aria-readonly': isReadOnly || undefined,
+      'aria-disabled': isDisabled || undefined,
+      'aria-orientation': 'horizontal',
+      'aria-valuetext': state.getValueLabelForIndex(index),
+      'aria-required': isRequired || undefined,
+      'aria-invalid': validationState === 'invalid' || undefined,
+      'aria-errormessage': thumbProps['aria-errormessage'],
+      onKeyDown: allowDrag ? draggableProps.onKeyDown : undefined
```

Comment:
I think you might need an `onChange` event handler here. Right now, adjusting the value with a screen reader isn't working. For example, on iOS you can use swipe gestures with VoiceOver to increment or decrement the value. This would fire change events on the hidden input.

You might not need the keyboard handler in that case - the native input might already handle the keyboard events and fire the onChange event accordingly. We'd need to test cross browser to ensure it's consistent.

Reply:
Great! Now the browser can also handle page-up/page-down as it sees fit